File: Back-End/ReviewService/src/main/java/com/ecommerce/reviewservice/event/ReviewEvent.java
package com.ecommerce.reviewservice.event;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReviewEvent {
    private UUID reviewId;
    private int productId;
    private UUID userId;
    private int rating;
    private String comment;
    private String action;  // "CREATED", "UPDATED", "DELETED"
    private LocalDateTime timestamp;
    private String performedBy;
}

File: Back-End/AdminLogsService/Events/ReviewEvent.cs
using System.Text.Json.Serialization;

namespace AdminLogsService.Events
{
    public class ReviewEvent
    {
        [JsonPropertyName("reviewId")]
        public long ReviewId { get; set; }
        
        [JsonPropertyName("productId")]
        public int ProductId { get; set; }
        
        [JsonPropertyName("userId")]
        public int UserId { get; set; }
        
        [JsonPropertyName("rating")]
        public int Rating { get; set; }
        
        [JsonPropertyName("comment")]
        public string? Comment { get; set; }
        
        [JsonPropertyName("action")]
        public string? Action { get; set; }
        
        [JsonPropertyName("timestamp")]
        public object? Timestamp { get; set; }  // Change to object to handle array format
        
        [JsonPropertyName("performedBy")]
        public string? PerformedBy { get; set; }
    }
}